src/pages/index.js

```diff
@@ -61,16 +107,64 @@ const addImageModal = new ModalWithForm(
 //-----------CREATE A MODALWITHIMAGE INSTANCE FOR THE PREVIEW IMAGE------------>>
 const previewModal = new ModalWithImage("#image-preview-modal");
 
+//-----------CREATE A FUNCTION THAT HANDLES DELETE CLICK------------>>
+function handleDeleteClick(card) {
+  deleteConfirmationModal.open();
+  deleteConfirmationModal.setCallback(() => {
+    api.deleteCard(card.getId()).catch((err) => console.log(err));
+    card.deleteCard();
+    deleteConfirmationModal.close();
+  });
+}
+
+//-----------CREATE A FUNCTION THAT HANDLES LIKE CLICK------------>>
+function handleLikeClick(card) {
+  api
+    .likeCard(card.getId(), card.isLiked)
+    .then((res) => card.toggleLikeCard(res.isLiked))
+    .catch((err) => console.log(err));
+}
+
+//-----------CREATE A FUNCTION THAT HANDLES AVATAR EDIT SUBMIT------------>>
+function handleAvatarFormSubmit(values) {
+  avatarEditModal.renderSaving(true);
+  api
+    .changeAvatar(values)
+    .then((res) => {
+      userInfo.setUserAvatar(res.avatar);
+    })
+    .catch((err) => console.log(err))
+    .finally(avatarEditModal.renderSaving(false));
+  avatarEditModal.close();
+}
+
 //-----------CREATE A FUNCTION THAT HANDLES PROFILE EDIT SUBMIT------------>>
 function handleProfileFormSubmit(values) {
-  userInfo.setUserInfo(values);
+  profileEditModal.renderSaving(true);
+  api
+    .editProfile(values)
+    .then((res) => {
+      userInfo.setUserInfo({
+        name: res.name,
+        description: res.about,
+      });
+    })
+    .catch((err) => console.log(err))
+    .finally(profileEditModal.renderSaving(false));
   profileEditModal.close();
 }
 
 //------------CREATE A FUNCTION THAT HANDLES ADD IMAGE SUBMIT---------->>
 function handleAddImageFormSubmit(values) {
-  const newCard = renderCard(values);
-  cardsContainer.addItem(newCard);
+  addImageModal.renderSaving(true);
+  api
+    .addCard(values)
+    .then((res) => {
+      const newCard = renderCard(res);
+      cardsContainer.addItem(newCard);
+    })
+    .catch((err) => console.log(err))
+    .finally(addImageModal.renderSaving(false));
   formValidators.addCardForm.resetForm();
   formValidators.addCardForm.disableSubmit();
   addImageModal.close();
```

Comment:
```js

  formValidators.addCardForm.resetForm();
  formValidators.addCardForm.disableSubmit();
```

This should be also only in `then` after a **successful response** from the server so that when there is an error we could send the data for the 2nd time